Lab-4/ejemyr_lab4.ipynb

```diff
@@ -0,0 +1,687 @@
+        "\n",
+        "# This file is part of the course DD2363 Methods in Scientific Computing\n",
+        "# KTH Royal Institute of Technology, Stockholm, Sweden\n",
+        "#\n",
+        "# This is free software: you can redistribute it and/or modify\n",
+        "# it under the terms of the GNU Lesser General Public License as published by\n",
+        "# the Free Software Foundation, either version 3 of the License, or\n",
+        "# (at your option) any later version."
+      ],
+      "execution_count": 3,
+      "outputs": [
+        {
+          "output_type": "execute_result",
+          "data": {
+            "text/plain": [
+              "'KTH Royal Institute of Technology, Stockholm, Sweden.'"
+            ]
+          },
+          "metadata": {
+            "tags": []
+          },
+          "execution_count": 3
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "28xLGz8JX3Hh",
+        "colab_type": "text"
+      },
+      "source": [
+        "# **Set up environment**"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "D2PYNusD08Wa",
+        "colab_type": "text"
+      },
+      "source": [
+        "To have access to the neccessary modules you have to run this cell. If you need additional modules, this is where you add them. "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "Xw7VlErAX7NS",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "# Load neccessary modules.\n",
+        "from google.colab import files\n",
+        "\n",
+        "import time\n",
+        "import numpy as np\n",
+        "\n",
+        "from matplotlib import pyplot as plt\n",
+        "from matplotlib import tri\n",
+        "from matplotlib import axes\n",
+        "from mpl_toolkits.mplot3d import Axes3D\n",
+        "from scipy.integrate import quad, dblquad"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "gnO3lhAigLev",
+        "colab_type": "text"
+      },
+      "source": [
+        "# **Introduction**"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "0DlTQl-x6dfJ",
+        "colab_type": "text"
+      },
+      "source": [
+        "In this lab we will investigate how to approximate functions in a 1d- and 2d-space. This will be done by finding the function $f_{\\pi} \\in P(I)$ such that $\\| f - f_{\\pi} \\|_{L^2}$ is minimized."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "WeFO9QMeUOAu",
+        "colab_type": "text"
+      },
+      "source": [
+        "# **Methods**"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "1WQUJfPg_drt",
+        "colab_type": "text"
+      },
+      "source": [
+        "## Approximation in 1D\n",
```

Comment:
You need to elaborate the theory here. It is unclear how you have reached your conclusions in regards to the number of integrals etc.

Reply:
Fixed it :) Sorry for the really late response to this...